source/src/BNFC/Backend/C.hs

```diff
@@ -348,12 +348,9 @@ bufferC bufferH = unlines
   , "    /* Make sure the buffer can hold all of s. */"
   , "    bufferAllocateChars(buffer, len);"
```

Comment:
I'm actually not clear if this is right, though it's not a *new* bug if so. Does `bufferAllocateChars` include a + 1 for the nul?

Reply:
The buffer `size` is always `>= 1` to hold at least the terminating null.  `bufferAllocateChars` makes sure that `len` extra characters have space.  There is no need to include the null in the count `len` because there is already room for the terminating null.
```c 
static void bufferAllocateChars (Buffer buffer, const unsigned int n) {
  /* 1 extra char for terminating 0. */
  unsigned int requiredSize = buffer->current + 1 + n;
...
```
TL;DR: Yes, it includes the +1, as you see.